server/src/main/java/org/elasticsearch/threadpool/Util.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.threadpool;
+
+import org.elasticsearch.common.unit.ByteSizeUnit;
+import org.elasticsearch.common.unit.ByteSizeValue;
+
+public class Util {
```

Comment:
What do you think of having these methods on BuiltinExecutorBuilders? Alternatively, could this be a more concrete name like ThreadPoolUtil?

Reply:
++ to have them in BuiltinExecutorBuilders

Reply:
Ok sure, let me try that. It has been a week and a half; I forget why I didn't do it that way in the first place.

Reply:
Alright, most of these methods are actually used exclusively by `DefaultBuildinExecutorBuilders` with the exception of two:
- `oneEighthAllocatedProcessors` is called by our old friend `Downsample`
- `searchOrGetThreadPoolSize` is called in a number of places:
  - `ESQL`
  - `QueryPragmas`
  - `TransportMultiSearchAction`

The ideal home for these methods depends, I think, on whether this is a pattern we want to encourage. Does it make sense for these callers to be calling these methods? If so, they probably belong someplace more "universal" than `BuiltinExecutorBuilders`.

Perhaps I could even put some or all of them back into `ThreadPool`; again, I forget why I moved them.

Reply:
Ok I moved the functions back where they were, in `ThreadPool`. That seems to work fine.

Reply:
I think in an earlier iteration of this code, I moved things into different packages, which made the utility methods inaccessible. But in the current code, they work just fine right where they are.